cclient/response/get_neighbors.c

```diff
@@ -11,14 +11,19 @@ void neighbor_info_t_copy(void* _dst, const void* _src) {
   neighbor_info_t *dst = (neighbor_info_t*)_dst, *src = (neighbor_info_t*)_src;
   dst->address = src->address;
   dst->all_trans_num = src->all_trans_num;
-  dst->invalid_trans_num = src->invalid_trans_num;
+  dst->random_trans_req_num = src->random_trans_req_num;
   dst->new_trans_num = src->new_trans_num;
+  dst->invalid_trans_num = src->invalid_trans_num;
+  dst->stale_trans_num = src->stale_trans_num;
+  dst->sent_trans_num = src->sent_trans_num;
+  dst->connectionType = src->connectionType;
 }
 
 void neighbor_info_t_dtor(void* _elt) {
   neighbor_info_t* elt = (neighbor_info_t*)_elt;
   if (elt->address) {
     char_buffer_free(elt->address);
+    char_buffer_free(elt->connectionType);
```

Comment:
sank_cast, also check if connectionType is NULL.

Reply:
char_buffer_free() would check if the input pointer is NULL.
Or you mean I should do something besides check if connectionType is NULL?

Reply:
right, `char_buffer_free` will check the pointer so you can remove if condition `if (elt->address)` as well.